lib/rex/proto/ms_dtyp.rb

```diff
@@ -0,0 +1,334 @@
+# -*- coding: binary -*-
+
+module Rex::Proto::MsDtyp
+    # [2.4.2.2 SID--Packet Representation](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/f992ad60-0fe4-4b87-9fed-beb478836861)
+    class MsDtypSid < BinData::Record
```

Comment:
FYI, there is [RPC_SID](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/5cb97814-a1c2-4215-b7dc-76d1f4bfad01) [implementation](https://github.com/rapid7/ruby_smb/blob/95ffce90f3fbd2b2d8d00b643e318fc38cce52bd/lib/ruby_smb/dcerpc/samr/rpc_sid.rb) in `ruby_smb` that also includes SID <-> text translation. It uses NDR structures, but it is quite similar. Just in case you wanted an implementation example.

Reply:
This work was taken from Spencer but I can see if we can potentially get that updated 👍

Reply:
Nice :) My idea was to actually query the LDAP server itself to get this information which will also allow it to be dynamic for any future updates or custom adjustments, though this is also an option.

Reply:
Added an initial implementation with f624a40 but noticed that well known SIDs don't come with good descriptions on DNs describing what they are actually used for which is a disappointment, so we might need a better solution for those.